controllers/packetmachine_controller.go

```diff
@@ -208,15 +210,22 @@ func (r *PacketMachineReconciler) reconcile(ctx context.Context, machineScope *s
 			return ctrl.Result{}, err
```

Comment:
The comment on this block is confusing because it is inverted.  "If we have a provider ID, fetch the device".

As far as I can tell, this is doing the right thing. Any error getting this device (404 or 5xx) will cause this device to try to reconcile again later. The first step of the reconcile is to check for 404s, so user-deleted (and failed to provision deleted) resources will be detected.

Reply:
> The comment on this block is confusing because it is inverted. "If we have a provider ID, fetch the device".

Indeed, fixed up the comment.

> As far as I can tell, this is doing the right thing. Any error getting this device (404 or 5xx) will cause this device to try to reconcile again later. The first step of the reconcile is to check for 404s, so user-deleted (also devices that failed-to-provision and were automatically deleted by EM) resources will be detected.

Ah, good point, I wasn't necessarily focused on this case with the changes I made, we should make it a fatal error if we get a 404 here, since it should be considered a fatal error from the CAPP perspective. The 404 check in `Reconcile` is related to Cluster API resources and not the EM api.